ElvargServer/src/main/java/com/elvarg/game/entity/impl/player/persistence/jsonfile/JSONFilePersistence.java

```diff
@@ -0,0 +1,91 @@
+package com.elvarg.game.entity.impl.player.persistence.jsonfile;
+
+import com.elvarg.Server;
+import com.elvarg.game.entity.impl.player.Player;
+import com.elvarg.game.entity.impl.player.persistence.PersistenceMethod;
+import com.elvarg.game.entity.impl.player.persistence.PlayerSave;
+import com.elvarg.util.Misc;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.logging.Level;
+
+public class JSONFilePersistence extends PersistenceMethod {
+
+    private static final String PATH = "./data/saves/characters/";
+
+    @Override
+    public PlayerSave retrieve(String username) {
+        if (!exists(username)) {
+            return null;
+        }
+
+        Path path = Paths.get(PATH, username + ".json");
+        File file = path.toFile();
+
+        try (FileReader fileReader = new FileReader(file)) {
+            JsonParser fileParser = new JsonParser();
+            Gson builder = new GsonBuilder().create();
+            JsonObject reader = (JsonObject) fileParser.parse(fileReader);
+
+            PlayerSave result = builder.fromJson(reader, PlayerSave.class);
```

Comment:
```suggestion
            JsonParser fileParser = new JsonParser();
            PlayerSave result = builder.fromJson(fileReader, PlayerSave.class);
```
Wont this just work?
And put `Gson builder = new GsonBuilder().create();` in final static field

Reply:
Yes works good, done!